api/config/service/k8s/v1/k8s.proto

```diff
@@ -8,4 +8,23 @@ import "validate/validate.proto";
 
 message Config {
   repeated string kubeconfigs = 1 [ (validate.rules).repeated = {unique : true} ];
+  RestClientConfig restclientconfig = 2;
+}
+
+// These configuration values are passed directly through to the rest config object.
+message RestClientConfig {
+  // The maximum length of time to wait before giving up on a server request.
+  // By default a value of zero means no timeout.
+  // https://github.com/kubernetes/client-go/blob/00dbcca6ee44c678754d3f5fda1bd0e704b26fe2/rest/config.go#L130-L131
+  uint32 timeout = 1 [(validate.rules).uint32.gte = 1];
```

Comment:
we'll almost always use duration to represent time (the underlying config uses duration as well)

```suggestion
  google.protobuf.Duration timeout = 1 [(validate.rules).duration.gt.seconds = 1];
```
(also needs import)